server/src/main/java/org/opensearch/index/store/lockmanager/LockManagerForClusterManagerNode.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import org.opensearch.index.store.RemoteBufferedOutputDirectory;
+import org.opensearch.index.store.RemoteSegmentStoreDirectory;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+
+public class LockManagerForClusterManagerNode extends RemoteStoreMetadataLockManager {
```

Comment:
given the duplicate classes and methods we need to create , I am inclined to create an instance of  `RemoteSegmentStoreDirectory` and calling `close()` on it .

Reply:
actually don't want to create RemoteSegmentStoreDirectory instance because RemoteSegmentStoreDirectory have lock manager instance, it could end up into a circular dependency problem.

Reply:
to remove some code duplication, creating the `RemoteSegmentStoreDirectory` inside `cleanupPostIndexDeletion` static method.

Reply:
creating `RemoteSegmentStoreDirectory` instance from `blobStoreRepository` now and then calling `close()` after releasing lock.